base/applications/cleanmgr/dialog.c

```diff
@@ -0,0 +1,301 @@
+    
+    switch (message)
+    {    
+        case WM_INITDIALOG:
+        {
+            BitmapDrive = LoadBitmapW(GetModuleHandleW(NULL), MAKEINTRESOURCEW(IDB_DRIVE));
+
+            if (BitmapDrive == NULL)
+            {
+                DPRINT("LoadBitmapW(): Failed to open the bitmap!\n");
+                EndDialog(hwnd, IDCANCEL);
+                return FALSE;
+            }
+
+            InitStartDlg(hwnd, BitmapDrive);
+            return TRUE;
+        }
+
+        case WM_NOTIFY:
+            return ThemeHandler(hwnd, (LPNMCUSTOMDRAW)lParam);
+
+        case WM_THEMECHANGED:
+            InvalidateRect(hwnd, NULL, FALSE);
+            break;
+
+        case WM_MEASUREITEM:
+        {
+            LPMEASUREITEMSTRUCT lpmis = (LPMEASUREITEMSTRUCT)lParam;
+
+            if (lpmis->itemHeight < 18 + 2)
+                lpmis->itemHeight = 18 + 2;
+
+            break;
+        }
+
+        case WM_DRAWITEM:
+        {
+            if (!DrawItemCombobox(lParam))
+            {
+                DPRINT("DrawItemCombobox(): Failed to initialize the ComboBox!\n");
+                EndDialog(hwnd, IDCANCEL);
+            }
+            break;
+        }
+
+        case WM_COMMAND:
+            if (HIWORD(wParam) == CBN_SELCHANGE)
+            {
+                HWND hComboCtrl = GetDlgItem(hwnd, IDC_DRIVE);
+                int ItemIndex = SendMessageW(hComboCtrl, CB_GETCURSEL, 0, 0);
+                StringCbCopyW(DriveLetter, sizeof(DriveLetter), GetProperDriveLetter(hComboCtrl, ItemIndex));
+            }
+
+            switch(LOWORD(wParam))
+            {
+                case IDOK:
+                    DeleteObject(BitmapDrive);
+                    EndDialog(hwnd, IDOK);
+                    break;
+                case IDCANCEL:
+                    EndDialog(hwnd, IDCANCEL);
+                    break;
+            }
+            break;
+
+        case WM_CLOSE:
+            EndDialog(hwnd, IDCANCEL);
+            break;  
+
+        case WM_DESTROY:
+            DeleteObject(BitmapDrive);
+            break;
+
+        default:
+        {
+            if (message == CleanmgrWindowMsg)
+            {
+                SetForegroundWindow(hwnd);
+            }
+            return FALSE;
+        }
+    }
+    return TRUE;
+}
+
+INT_PTR CALLBACK ProgressDlgProc(HWND hwnd, UINT message, WPARAM wParam, LPARAM lParam)
+{   
+    switch(message)
+    {
+        case WM_INITDIALOG:
+        {
+            WCHAR FullText[ARR_MAX_SIZE] = { 0 };
+            WCHAR SysDrive[ARR_MAX_SIZE] = { 0 };
+            WCHAR TempText[ARR_MAX_SIZE] = { 0 };
+            HANDLE ThreadOBJ = NULL;
+
+            IsSystemDrive = FALSE;
+            LoadStringW(GetModuleHandleW(NULL), IDS_SCAN, TempText, _countof(TempText));
+            StringCchPrintfW(FullText, sizeof(FullText), TempText, DriveLetter);
+            SetDlgItemTextW(hwnd, IDC_STATIC_SCAN, FullText);
+            GetEnvironmentVariableW(L"SystemDrive", SysDrive, sizeof(SysDrive));
```

Comment:
Bad way of reliably getting the system drive, if I manually run the cleanmgr in a cmd prompt where I manually modified this environment variable.